Gems/Atom/RPI/Code/Source/RPI.Edit/Material/MaterialTypeSourceData.cpp

```diff
@@ -1035,7 +1055,70 @@ namespace AZ
             return true;
         }
 
-        Outcome<Data::Asset<MaterialTypeAsset>> MaterialTypeSourceData::CreateMaterialTypeAsset(Data::AssetId assetId, AZStd::string_view materialTypeSourceFilePath, bool elevateWarnings) const
+        MaterialShaderParameterLayout MaterialTypeSourceData::CreateMaterialShaderParameterLayout() const
+        {
+            // Create the MaterialParameter-Struct-definition from the properties that have a shader connection
+            MaterialShaderParameterLayout shaderParamsLayout{};
+            // the first entry is always the material type
+            shaderParamsLayout.AddMaterialParameter<uint32_t>("m_materialType", true);
+            // TODO: remove this (debug only)
```

Comment:
Should this be removed before merging?

Reply:
i'm actually verifying the instance id on the gpu, so i've changed the comment